toonz/sources/toonz/startuppopup.cpp

```diff
@@ -0,0 +1,889 @@
+  m_widthLabel              = new QLabel(tr("Width:"), this);
+  m_widthFld                = new DVGui::MeasuredDoubleLineEdit(0);
+  m_heightLabel             = new QLabel(tr("Height:"), this);
+  m_heightFld               = new DVGui::MeasuredDoubleLineEdit(0);
+  m_dpiLabel                = new QLabel(tr("DPI:"), this);
+  m_dpiFld                  = new DoubleLineEdit(0, 120);
+  m_resXLabel               = new QLabel(tr("X"), this);
+  m_resXFld                 = new DVGui::DoubleLineEdit(this);
+  m_resYFld                 = new DVGui::DoubleLineEdit(this);
+  m_resTextLabel            = new QLabel(tr("Resolution:"), this);
+  m_fpsLabel                = new QLabel(tr("Frame Rate:"), this);
+  m_fpsFld                  = new DoubleLineEdit(0, 24.0);
+  m_cameraSettingsWidget    = new CameraSettingsWidget(false);
+  m_presetCombo             = new QComboBox(this);
+  m_unitsCB                 = new QComboBox(this);
+  m_addPresetBtn            = new QPushButton(tr("Add"), this);
+  m_removePresetBtn         = new QPushButton(tr("Remove"), this);
+  m_showAtStartCB           = new QCheckBox(tr("Show this at startup"), this);
+  QPushButton *createButton = new QPushButton(tr("Create Scene"), this);
+  QPushButton *newProjectButton = new QPushButton(tr("New Project..."), this);
+  QPushButton *loadOtherSceneButton =
+      new QPushButton(tr("Open Another Scene..."), this);
+  m_projectsCB = new QComboBox(this);
+  QStringList type;
+  type << tr("pixel") << tr("cm") << tr("mm") << tr("inch") << tr("field");
+  m_unitsCB->addItems(type);
+
+  // Exclude all character which cannot fit in a filepath (Win).
+  // Dots are also prohibited since they are internally managed by Toonz.
+  QRegExp rx("[^\\\\/:?*.\"<>|]+");
+  m_nameFld->setValidator(new QRegExpValidator(rx, this));
+
+  m_widthFld->setMeasure("camera.lx");
+  m_heightFld->setMeasure("camera.ly");
+
+  m_widthFld->setRange(0.1, (std::numeric_limits<double>::max)());
+  m_heightFld->setRange(0.1, (std::numeric_limits<double>::max)());
+  m_fpsFld->setRange(1.0, (std::numeric_limits<double>::max)());
+  m_dpiFld->setRange(1.0, (std::numeric_limits<double>::max)());
+  m_resXFld->setRange(0.1, (std::numeric_limits<double>::max)());
+  m_resYFld->setRange(0.1, (std::numeric_limits<double>::max)());
+  m_showAtStartCB->setChecked(Preferences::instance()->isStartupPopupEnabled());
+  m_showAtStartCB->setStyleSheet("QCheckBox{ background-color: none; }");
+  m_addPresetBtn->setStyleSheet(
+      "QPushButton { padding-left: 4px; padding-right: 4px;}");
+  m_removePresetBtn->setStyleSheet(
+      "QPushButton { padding-left: 4px; padding-right: 4px;}");
+  QLabel *label = new QLabel();
+  label->setPixmap(QPixmap(":Resources/startup.png"));
+  label->setObjectName("StartupLogo");
+  m_projectBox->setObjectName("StartupProjectBox");
+  m_sceneBox->setObjectName("StartupNewSceneBox");
+  m_recentBox->setObjectName("StartupRecentSceneBox");
```

Comment:
If these three boxes can be the same design, I think it would be clear to give them the same object name.
Just sharing `SolidLineFrame` would be fine as it is already defined in the style sheet.